server/game/cards/02_SHD/units/TheMandalorian.ts

```diff
@@ -0,0 +1,34 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { AbilityContext } from '../../../core/ability/AbilityContext';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { WildcardCardType } from '../../../core/Constants';
+
+export default class TheMandalorian extends NonLeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '6585115122',
+            internalName: 'the-mandalorian#wherever-i-go-he-goes',
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.addWhenPlayedAbility({
+            title: 'You may heal all damage from a unit that costs 2 or less and give 2 Shield tokens to it',
+            optional: true,
+            targetResolver: {
+                cardTypeFilter: WildcardCardType.Unit,
```

Comment:
Can omit this since we have `card.isUnit()` in the card condition anyway

Reply:
I removed the cardTypeFilter, but then i started getting this assertion error that i tried, but couldn't resolve yet

`AssertionError: Contract assertion failure: Attempting to read property 'damage' on 'darth-vader#dark-lord-of-the-sith' but it is in location 'base' where the property does not apply`

Reply:
Ah sorry, I know what that is. I just merged a bug fix that I _think_ will resolve it, please merge main when you're ready and try again.

Reply:
its done